src/components/appheader.main.jsx

```diff
@@ -183,29 +187,34 @@ class AppHeaderMain extends React.Component {
         </div>
 
         <div className="collapsable-container collapse collapsed">
-          <div className="search-container">
-            <AppHeaderSearchMain isMobileView isFocused={isSearchFocused} />
-          </div>
+          {!isB2B && !Config.b2b.enable && (
+            <div className="search-container">
+              <AppHeaderSearchMain isMobileView isFocused={isSearchFocused} />
+            </div>
+          )}
           <div className="mobile-locale-container">
             <AppHeaderLocaleMain isMobileView />
           </div>
-          <div className="mobile-cart-link-container">
-            <Link
-              className="cart-link"
-              to="/mybag"
-            >
-              <div data-toggle="collapse" data-target=".collapsable-container">
-                {intl.get('shopping-bag-nav')}
-                <div className="cart-link-counter-container">
-                  {cartData && cartData['total-quantity'] !== 0 && !isLoading && (
-                    <span className="cart-link-counter">
-                      {cartData['total-quantity']}
-                    </span>
-                  )}
+
+          {!isB2B && !Config.b2b.enable && (
```

Comment:
This condition won't work as expected. We need to show these components only when we're in B2C, or only in B2B after a buyer has logged in.

Something like the following should work better:
`{!Config.b2b.enable || (Config.b2b.enable && isB2B) && (....`